lxd/storage/drivers/driver_btrfs_volumes.go

```diff
@@ -1338,9 +1416,35 @@ func (d *btrfs) UnmountVolumeSnapshot(snapVol Volume, op *operations.Operation)
 	return forceUnmount(snapPath)
 }
 
-// VolumeSnapshots returns a list of snapshots for the volume.
+// VolumeSnapshots returns a list of snapshots for the volume in age order (oldest first).
 func (d *btrfs) VolumeSnapshots(vol Volume, op *operations.Operation) ([]string, error) {
-	return genericVFSVolumeSnapshots(d, vol, op)
```

Comment:
@stgraber the `dir` and `cephfs` drivers also use this function `genericVFSVolumeSnapshots` to get their snapshot names on the storage layer.

I can imagine this causing problems with `cephfs` snapshot efficiency, and while it doesn't cause issues directly for `dir` volumes, it will be an issue when migrating a dir volume to a non-dir pool that does have proper snapshot support.

Maybe cephfs has a native way of getting snapshots by oldest first (like I've switched to here for BTRFS), but for dir, this could be tricky as we cannot rely on the create/update time of the snapshot directories.

Reply:
@stgraber ive created https://github.com/lxc/lxd/pull/8550 which avoids needing to change the "public" storage interface, but instead changes the driver interface to accept a slice of snapshot names, and then backendLXD performs the lookup from the database.